python/triton/language/semantic.py

```diff
@@ -476,10 +476,16 @@ def not_equal(input: tl.tensor,
 
 
 def arange(start: int, end: int, builder: ir.builder) -> tl.tensor:
-    if not isinstance(start, int) or not isinstance(end, int):
+    if not isinstance(start, int) or not isinstance(end, int) or start >> 32 or end >> 32:
```

Comment:
I think we should be explicit here, and have something like: `is_start_int64 = bool(start >> 32)`

Reply:
done